### PR TITLE
Decision date validation nonrating request issues

### PR DESCRIPTION
**Pull request by Sjones352**
Connects  #10714
Screen Shot 2019-05-17 at 5 18 18 PM

### CODE REVIEW

File: spec/support/intake_helpers.rb [hunk 1]
```diff
@@ -206,6 +206,7 @@ def add_intake_nonrating_issue(
     legacy_issues: false
   )
     add_button_text = legacy_issues ? "Next" : "Add this issue"
+    sleep 1
```
Comment (pkarman) [inline spec/support/intake_helpers.rb]: why is this required?

File: spec/feature/intake/higher_level_review_spec.rb [hunk 2]
```diff
@@ -736,6 +736,20 @@ def complete_higher_level_review
...
+        visit "/intake/add_issues"
+        click_intake_add_issue
+
+        fill_in "Issue category", with: "Apportionment"
+        find("#issue-category").send_keys :enter
+
+        fill_in "Decision date", with: "13/04/2019"
+
+        err_message = "Please enter a valid decision date"
+        expect(err_message).to_not eq nil
```
Comment (pkarman) [inline spec/feature/intake/higher_level_review_spec.rb]: this test will always pass. were you trying to test `expect(page).to_not have_content(err_message)` instead?

File: client/app/intake/util/issues.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ export const legacyIssue = (issue, legacyAppeals) => {
 };
 
 export const validateDate = (date) => {
-  const datePattern = /^(0[1-9]|1[0-2])[/](0[1-9]|[12][0-9]|3[01])[/](19|20)\d\d$/;
+  const datePattern = /^(0[1-9]|1[0-2])[/](0[1-9]|[12][0-9]|3[01])[/](19|20)?\d\d$/;
```
Comment (pkarman) [inline client/app/intake/util/issues.js]: adding this `?` makes the final year optionally 2-digits or 4. Is that what we want?

File: spec/support/intake_helpers.rb (hunk 1 above)
Comment (Sjones352) [reply spec/support/intake_helpers.rb]: Removed

File: spec/feature/intake/higher_level_review_spec.rb (hunk 2 above)
Comment (Sjones352) [reply spec/feature/intake/higher_level_review_spec.rb]: Fixed

File: client/app/intake/util/issues.js (hunk 3 above)
Comment (Sjones352) [reply client/app/intake/util/issues.js]: Removed

File: spec/feature/intake/intake_edit_confirmation_spec.rb [hunk 4]
```diff
@@ -64,7 +64,7 @@
             # first add a nonrating issue so we can remove the rating issue & EP
             click_intake_add_issue
             click_intake_no_matching_issues
-            add_intake_nonrating_issue(date: (decision_review.receipt_date - 1.month).strftime("%D"))
+            add_intake_nonrating_issue(date: (decision_review.receipt_date - 1.month).strftime("%m/%d/%Y"))
```
Comment (pkarman) [inline spec/feature/intake/intake_edit_confirmation_spec.rb]: you can use the `.mdY` method here in tests instead of `strftime`.

File: spec/feature/intake/higher_level_review_spec.rb [hunk 5]
```diff
@@ -726,6 +726,20 @@ def complete_higher_level_review
...
+        start_higher_level_review(veteran_no_ratings)
+        visit "/intake/add_issues"
+        click_intake_add_issue
+
+        fill_in "Issue category", with: "Apportionment"
+        find("#issue-category").send_keys :enter
+
+        fill_in "Decision date", with: "13/04/2019"
+
+        err_message = "Please enter a valid decision date"
```
Comment (leikkisa) [inline spec/feature/intake/higher_level_review_spec.rb]: Just a style nit, I don't think you need to put this into it's own variable.

File: spec/feature/intake/higher_level_review_spec.rb [hunk 6]
```diff
@@ -726,6 +726,20 @@ def complete_higher_level_review
 
         expect(page).to have_content("1 issue")
       end
+
+      scenario "validate decsion date" do
```
Comment (leikkisa) [inline spec/feature/intake/higher_level_review_spec.rb]: This has a typo

File: client/app/intake/components/NonratingRequestIssueModal.jsx [hunk 7]
```diff
@@ -208,6 +211,7 @@ class NonratingRequestIssueModal extends React.Component {
             label="Decision date"
             strongLabel
             value={decisionDate}
+            errorMessage={this.state.dateError ? 'Please enter a valid decision date.' : null}
```
Comment (leikkisa) [inline client/app/intake/components/NonratingRequestIssueModal.jsx]: This could go into the copy file, but I don't feel strongly about it.
